### PR TITLE
Add leapwallet metamask cosmos snap support

### PR DESCRIPTION
**Pull request by baryon2**
Adds metamask cosmos snap support.

### CODE REVIEW

**Comment by pyramation** [comment]
I noticed the metamask icon — do you guys have permission to use it?

**Comment by pyramation** [comment]
additionally, if we use metamask icon, we need permission or it should represent metamask, not a snap I would imagine. Can you guys resolve this somehow? Then we can merge :)

**Comment by liujun93** [comment]
@baryon2 description field has been added. pull the latest main please.

**Comment by baryon2** [comment]
@liujun93 I have updated the PR with the description.

File: wallets/leap-metamask-cosmos-snap/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from './extension';
+export * from './leap';
```
Comment (sgup) [inline wallets/leap-metamask-cosmos-snap/src/index.ts:2]: This import needs to be updated to `./leap-metamask-cosmos-snap`

**Comment by pyramation** [comment]
before we merge, let's also add a new field somehow to indicate that this is a snap 
cc @HebeHe1986 

I think we will have dozens of snaps and need to organize/filter them

**Comment by pyramation** [comment]
we have a few items in this epic here first https://github.com/cosmology-tech/cosmos-kit/issues/301

**Comment by pyramation** [comment]
I propose we use the official metamask logo, but we superposition company/snap logos onto it to minimize confusion: https://github.com/cosmology-tech/cosmos-kit/issues/303

we'll have a product dilemma as we intend to directly support metamask, and using the metamask logo alone could create confusion. The keplr dashboard does a nice job of this with IBC assets

**Comment by pyramation** [comment]
since it's an isolated module, I'll merge now, but we'll address these issues/comments asap, thanks :)

**Comment by pyramation** [comment]
you can test here 

```
Successfully published:
 - @cosmos-kit/leap-metamask-cosmos-snap@0.0.2
 ```

**Comment by liujun93** [comment]
Hi, @baryon2  Some user is asking for `addChain` implementation in `leap-metamask-cosmos-snap`. Would you be able to implement it in the snap client?

**Pull request merged**